test/performance/scale_revision_by_load_test.go

```diff
@@ -121,11 +122,11 @@ func scaleRevisionByLoad(t *testing.T, numClients int) []junit.TestCase {
 	}
 	t.Logf("Took %v for the endpoint to start serving", time.Since(st))
 
-	// The number of scale events should be at most ~numClients/targetConcurrency,
-	// adding a big buffer to account for unexpected events
-	scaleCh := make(chan *scaleEvent, numClients/targetConcurrency*10)
+	// The number of scale events should be at most ~numClients/targetConcurrency
+	scaleEvents := make([]*scaleEvent, 0, numClients/targetConcurrency*10)
 	stopCh := make(chan struct{})
 
+	var m sync.Mutex
```

Comment:
please place this right next to `scaleEvents` and give a more speaking name, i.e. `scaleEventsMutex` so we know what it's locking.